src/OpenTelemetry/Metrics/Exemplar/Exemplar.cs

```diff
@@ -23,6 +23,8 @@ namespace OpenTelemetry.Metrics
     /// </summary>
     public struct Exemplar
     {
+        internal List<KeyValuePair<string, object>> AllTags;
```

Comment:
i just realized we dont need 2 separate lists. one is fine (FilteredTags). It can store all tags internally, and during snaptshot, it can be modified to store the actuall FilteredTags by FIlteredTags - FilteredTags - StoredTags